streams/src/test/java/org/apache/kafka/streams/integration/NamedTopologyIntegrationTest.java

```diff
@@ -924,8 +926,10 @@ private void verifyPartitionsAndStoresForTopology(final String topologyName, fin
                                                             .collect(Collectors.toList())),
             is(true));
 
-        // then verify that only this topology's one store appears
-        assertThat(metadata.stateStoreNames(), equalTo(singleton("store-" + topologyName)));
+        // then verify that only this topology's one store appears if the host has partitions assigned
+        if (!metadata.topicPartitions().isEmpty()) {
+            assertThat(metadata.stateStoreNames(), equalTo(singleton("store-" + topologyName)));
+        }
```

Comment:
It seems like this test is generally missing the converse:

In addition to checking if all the assigned topics are in the named topology sources, should we also check that all named topology sources are assigned?

Likewise, in this new block, should we assert that the store is not present if the partition is not assigned?

Reply:
Well, not all named topology topics are going to be in all the source metadata, we can check the stores are limited to the relevant topics. Though I am not sure how useful that is as the method for getting the metadata is filtering the store by if they contain the topology name.